README.md

```diff
@@ -385,6 +399,137 @@ jsn:select({value, [<<"user">>, <<"id">>]},
 % [<<"456">>]
 ```
 
+#### <a name="with-without"/>Using `with/2` and `without/2`
+
+The functions `with/2` and `without/3` accept a list of `path()`s or a
+`path_elements_map()` defined in [jsn.hrl](include/jsn.hrl). These functions
+are inspired by [`maps:with/2`](https://erlang.org/doc/man/maps.html#with-2) and
+[`maps:without/2`](https://erlang.org/doc/man/maps.html#without-2), but support
+the nested path formats of this library. Essentially, you can pass a list of
+arbitrarily nested paths to either function, and they will return your input
+with those paths applied, either by including **only** those paths in the input
+in the case of `with/2`, or removing those paths from the input in the case of
+`without/2`.
+
+However, because nested paths are considerably more complex than working with
+the simple keys list of the `maps` analogues of these functions, there are some
+caveats and specific aspects to be aware of when using these functions.
+
+* For `with/2`, inclusion happens at the first leaf node encountered when
+  traversing the given object and paths. That is to say, if you pass in a deeply
+  nested path, such as `foo.bar.baz.fum`, but the object only has a sub-path
+  of that path, say, `foo.bar.baz = "not an object`, the sub-path will be
+  included.
```

Comment:
I don't find this example to be helpful. Do you mean to say that if the object has a path like `foo.bar.baz` and the code is searching for `foo.bar.baz.fum` that the key `foo.bar.baz` _will_ be included in the output?

Reply:
That's exactly what I am saying.

Reply:
Did I misunderstand the desired behavior here, BTW?

```erlang
Object = #{<<"foo">> => #{<<"bar">> => "not an object"}}.
% #{<<"foo">> => #{<<"bar">> => "not an object"}}

jsn:with(['foo.bar.baz'], Object).
% (1) don't include anything because no exact match: 
%     #{}
% or
% (2) include the partial path:
%    #{<<"foo">> => #{<<"bar">> => "not an object"}}
```

Did I misunderstand your advice earlier? If you think we should go with (1) above, I'll fix the implementation to do that and I think we can remove both of these comments, since the behavior would be consistent (only exact match of paths trigger inclusion/exclusion).

Reply:
That right. I think paths should **exactly** match to be included. Otherwise, they are ignored. Partial matches are not **exact** and so should be ignored. To me, that's what the principle of least surprise mandates.

If you want to have like `fuzzy_with` and `fuzzy_without` then the semantics you have now would make sense. Maybe if you're basic you're going to call the functions `partial_path_match_with` and `partial_path_match_without`

Reply:
And honestly, as long as you're allowing partial matches in a path, you might as well go full regex for key spec imo.

Reply:
Fair enough, on reflection I totally agree with you. I've updated the code to behave accordingly (and removed these README points).